roachtest: more informatively tear down after timeout

tbg opened the pull request:
When a test times out and fails to respond to cancellation, we already
dump the roachtest stacks (=test side of the hang).

With this commit, we additionally dump stacks for any processes
running on remote machines (=server side of the hang).

Additionally, this might improve our odds at actually nudging the test
to terminate than we did previously. (Though the context cancellation
ought to also have killed all processes).

Release justification: roachtest change only
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45887)
<!-- Reviewable:end -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45887#-:-M23mj4V-4Aid4RlJYZ4:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)



<!-- Sent from Reviewable.io -->

tbg committed:
roachtest: more informatively tear down after timeout

When a test times out and fails to respond to cancellation, we already
dump the roachtest stacks (=test side of the hang).

With this commit, we additionally dump stacks for any processes
running on remote machines (=server side of the hang).

Additionally, this might improve our odds at actually nudging the test
to terminate than we did previously. (Though the context cancellation
ought to also have killed all processes).

Release justification: roachtest-only change
Release note: None

tbg commented:
bors r=nvanbenschoten

tbg commented:
there was a release justification. Let's try again with one added to PR

bors r=nvanbenschoten
